### ISSUE TITLE
Determine State of Motion & Fitness API

### ISSUE THREAD

**Issue by brandonhub** [transistorsoft/flutter_background_geolocation#153]
## My Environment
* **Version**: 1.3.2
* **Platform**: iOS or Android iOS
* **OS version**: Latest (13.1)
* **Device manufacturer / model**: iPhone 8
* **Flutter info (`flutter doctor`)**: 
```
[✓] Flutter (Channel stable, v1.9.1+hotfix.6, on Mac OS X 10.15.1 19B88, locale en-US)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 11.2.1)
[✓] Android Studio (version 3.4)
[✓] VS Code (version 1.40.1)
[!] Connected device
    ! No devices available
```
* Plugin config:
```dart <-- Syntax highlighting: DO NOT REMOVE -->
Config(
      desiredAccuracy: Config.DESIRED_ACCURACY_HIGH,
      stopOnTerminate: false,
      startOnBoot: true,
      logLevel: Config.LOG_LEVEL_VERBOSE,
      url: serverUrl,
      autoSyncThreshold: 25,
      batchSync: true,
      preventSuspend: true,
      params: {'userID': <userID goes here>})
```

## Expected Behavior
For there to be some way to tell if the Motion/Fitness API is enabled.

## Actual Behavior
There is no way to tell if the Motion/Fitness API is enabled.

## Context
This plugin's documentation stipulates a strong emphasis on the Motion/Fitness API to allow for battery-efficient location tracking. It would be great if there was a way we could get the status of this permission so that we can report to the user if is not enabled. I know that the plugin will request the permission the first time `BackgroundGeolocation.start` (or an equivalent) is called but that isn't enough to determine if the API is enabled or not.

Awesome plugin btw!

**Comment by christocracy** [comment]
Initially, Apple didn’t make it easy to determine this.  You actually had to attempt to turn on updates and check for an error ref != nil.

I think they added an api to determine permission in iOS 11 or 12.

**Comment by robrichardson13** [comment]
In iOS this function should work to check if they have motion enabled or not. It would be great to get this into the plugin!

`    func hasMotionPermissions() -> Bool {
        if #available(iOS 11.0, *) {
            let status = CMSensorRecorder.authorizationStatus()
            return status == .authorized
        } else {
            return CMSensorRecorder.isAuthorizedForRecording()
        }
    }`
